### PR TITLE
FilesRecyclerView improvements

### PR DESCRIPTION
**Pull request by jacobh-gh**
- Refactor the Files RecyclerView use a readonly MvxObservableCollection for the data structure for sky files, and use SwitchTo() as default way of updating the list

- FilesViewModel refactoring changes to add on previous changed & add some useful members to the class

- Files list selection action is now toggleable

- Fixed some RecyclerView bugs where the list would not update correctly after adding new items or some other update
 
- I don't know if it was broken before I updated the collections, but after removing unnecessary uses of RaisePropertyChanged() on the list's data source it created issues. 

After these changes, the FilesRecyclerView is more optimised, by performing less expensive updates than before.

### CODE REVIEW

**Comment by jacobh-gh** [comment]
@tomShoutTheSecond before merging, these changes broke the isNew feature somewhat. There can now be more than one new item in the list if you upload more than one new item. Can you see where I broke it?

**Comment by tomShoutTheSecond** [comment]
> @tomShoutTheSecond before merging, these changes broke the isNew feature somewhat. There can now be more than one new item in the list if you upload more than one new item. Can you see where I broke it?

This sounds like the expected behaviour to me, it was designed to highlight all files that were uploaded during the current session with the IsNew property, to distinguish them from previously uploaded files. This might change when we redesign the Files UI

File: SkyDrop/src/SkyDrop.Core/ViewModels/FilesViewModel.cs [hunk 1]
```diff
@@ -129,23 +143,25 @@ private async Task CopyFileLinkToClipboard(SkyFile skyFile)
...
-                StagedFiles = new List<StagedFileDVM>(StagedFiles);
-                _ = RaisePropertyChanged(() => StagedFiles);
+                var newSkyFiles = new List<StagedFileDVM>(StagedFiles);
+                StagedFiles.SwitchTo(newSkyFiles);
 
                 await UploadFile(stagedFile.StagedFile);
 
                 stagedFile.IsLoading = false;
-                StagedFiles = new List<StagedFileDVM>(StagedFiles.Where(f => f.StagedFile.Filename != stagedFile.StagedFile.Filename));
-                _ = RaisePropertyChanged(() => StagedFiles);
```
Comment (tomShoutTheSecond) [inline SkyDrop/src/SkyDrop.Core/ViewModels/FilesViewModel.cs:144]: ProgressBars are now not showing on file cells when uploading files, could be related to removing this call. Also possibly related to FodyWeaver setup.

File: SkyDrop/src/SkyDrop.Core/ViewModels/FilesViewModel.cs [hunk 2]
```diff
@@ -129,23 +143,25 @@ private async Task CopyFileLinkToClipboard(SkyFile skyFile)
...
+
+                StagedFiles.Remove(stagedFile);
             }
 
             IsLoading = false;
-            _ = RaisePropertyChanged(() => IsLoading);
```
Comment (tomShoutTheSecond) [inline SkyDrop/src/SkyDrop.Core/ViewModels/FilesViewModel.cs:148]: ProgressBar on Upload button is now not showing, could be related to removing this call (although FodyWeaver should handle this properly, something wrong with FodyWeaver config possibly)

**Comment by jacobh-gh** [comment]
fixed fody weaver

**Pull request merged**